src/types/slices.ts

```diff
@@ -0,0 +1,131 @@
+import { AvailableTabs, Limits } from 'appConstants';
+import {
+  PokemonMoveDetails,
+  PokemonTypeDetails,
+  PokemonBaseData,
+  PokemonDetails,
+  PokemonSorting,
+  TypeBaseData,
+  MoveBaseData,
+  MoveSorting,
+  TypeSorting,
+  CustomPokemon,
+} from 'types';
+
+export type ResourcesState = {
+  searchTypeResults: Array<PokemonTypeDetails> | null;
+  searchMoveResults: Array<PokemonMoveDetails> | null;
+  searchPokemonResults: Array<PokemonDetails> | null;
+  currentMoves: Array<PokemonMoveDetails>;
+  currentTypes: Array<PokemonTypeDetails>;
+  currentPokemons: Array<PokemonDetails>;
+  status: Status;
+  basePokemons: PokemonBaseData | null;
+  pokemonResultsAmount: Limits.pokemon;
+  moveResultsAmount: Limits.move;
+  typeResultsAmount: Limits.type;
+  allPokemons: Array<PokemonDetails>;
+  allMoves: Array<PokemonMoveDetails>;
+  allTypes: Array<PokemonTypeDetails>;
+  baseMoves: MoveBaseData | null;
+  baseTypes: TypeBaseData | null;
+  pokemonSorting: PokemonSorting;
+  currentPokemonPage: number;
+  currentMovePage: number;
+  currentTypePage: number;
+  moveSorting: MoveSorting;
+  typeSorting: TypeSorting;
+};
+
+export enum Status {
+  loading = 'loading',
+  failed = 'failed',
+  idle = 'idle',
+}
+
+export type SortingThunkParams = {
+  resourceType: 'pokemons' | 'moves' | 'types';
+  resultsAmount: number;
+};
+
+export type ResultsAmountParams = {
+  resourceType: 'pokemons' | 'moves' | 'types';
+  resultsAmount: number;
+};
+
+export type NextPageParams = {
+  resourceType: 'pokemons' | 'moves' | 'types';
+  next: string;
+};
+
+export type PreviousPageParams = {
+  resourceType: 'pokemons' | 'moves' | 'types';
+  previous: string;
+};
+
+export type SpecificPageParams = {
+  resourceType: 'pokemons' | 'moves' | 'types';
```

Comment:
```suggestion
  resourceType: AvailableTabs;
```
there and all other such cases you can use enum as a type.

Reply:
Thanks :)